newton/examples/ik/example_ik_custom.py

```diff
@@ -0,0 +1,324 @@
+            ("fr3_link6", 8, 0.06),  # wrist housing: catches close passes near fixtures when orienting the tool
+            # Optional but helpful if space is tight around the shoulder/torso:
+            ("fr3_link2", 4, 0.075),  # upper arm near shoulder: guards early sweep during large reorientations
+        ]
+
+        # ------------------------------------------------------------------
+        # Persistent gizmo transforms (pass-by-ref objects mutated by viewer)
+        # ------------------------------------------------------------------
+        body_q_np = self.state.body_q.numpy()
+        self.ee_tf = wp.transform(*body_q_np[self.ee_index])
+        self.sphere_tf = wp.transform(self.obstacle_center, wp.quat_identity())
+
+        # ------------------------------------------------------------------
+        # IK setup
+        # ------------------------------------------------------------------
+        def _q2v4(q):
+            return wp.vec4(q[0], q[1], q[2], q[3])
+
+        # Position & rotation objectives ------------------------------------
+        self.pos_obj = ik.IKPositionObjective(
+            link_index=self.ee_index,
+            link_offset=wp.vec3(0.0, 0.0, 0.0),
+            target_positions=wp.array([wp.transform_get_translation(self.ee_tf)], dtype=wp.vec3),
+        )
+
+        self.rot_obj = ik.IKRotationObjective(
+            link_index=self.ee_index,
+            link_offset_rotation=wp.quat_identity(),
+            target_rotations=wp.array([_q2v4(wp.transform_get_rotation(self.ee_tf))], dtype=wp.vec4),
+        )
+
+        # Collision objectives ----------------------------------------------
+        # Share the same arrays across all objectives so one update suffices.
+        self.obstacle_centers = wp.array([self.obstacle_center], dtype=wp.vec3)
+        self.obstacle_radii = wp.array([self.obstacle_radius], dtype=wp.float32)
+
+        self.collision_objs = []
+        for _, (_, link_idx, link_radius) in enumerate(self.links_to_check_collision):
+            self.collision_objs.append(
+                CollisionSphereAvoidObjective(
+                    link_index=link_idx,
+                    link_offset=wp.vec3(0.0, 0.0, 0.0),
+                    link_radius=link_radius,
+                    obstacle_centers=self.obstacle_centers,
+                    obstacle_radii=self.obstacle_radii,
+                    weight=10.0,
+                )
+            )
+
+        # Joint limit objective (starts after collisions)
+        self.obj_joint_limits = ik.IKJointLimitObjective(
+            joint_limit_lower=self.model.joint_limit_lower,
+            joint_limit_upper=self.model.joint_limit_upper,
+            weight=10.0,
+        )
+
+        # Variables the solver will update
+        self.joint_q = wp.array(self.model.joint_q, shape=(1, self.model.joint_coord_count))
+
+        self.ik_iters = 10
+        self.solver = ik.IKSolver(
```

Comment:
Maybe rename this to `ik_solver` since we've named the physics solver as `solver` in all the other examples.